astropy/modeling/tests/test_mappings.py

```diff
@@ -86,3 +86,13 @@ def test_fittable_compound():
     new_model = pfit(m, x, y_noisy)
     y_fit = new_model(x)
     assert_allclose(y_fit, y_real, atol=dy)
+
+
+def test_identity_repr():
+    m = Identity(1, name='foo')
+    assert repr(m) == "<Identity(1, name='foo')>"
```

Comment:
Why not also add number and whatever special char name is supposed to support, just to be sure?

Reply:
I'm not sure what you mean about number and special char?  Name must be a `str`.

Reply:
orly? So it cannot be `'MyGaussian1D'` or `'My_Gaussian1D'`?